### PR TITLE
Closing #1467 new chat modal 2 page layout

### LINKED ISSUE #1467: New chat: 2-page layout
https://www.figma.com/file/EJxXOzem02zhRvuqsJIool/Actual-chat-Desktop?type=design&node-id=11427-302978&mode=design&t=i1I3uy57tXfKKGc2-0

https://dev.actual.chat/chat/052w3sgrad#44387

### PR DESCRIPTION
**Pull request by crui3er**
New chat modal with 2 pages:
image
image

### CODE REVIEW

File: src/dotnet/Chat.Service/ChatsBackend.cs [hunk 1]
```diff
@@ -328,9 +328,14 @@ await peerChatId.UserIds
             else if (chatId.Kind == ChatKind.Group) {
                 // Group chat
                 ownerId.Require("Command.OwnerId");
-                var author = await AuthorsBackend
-                    .EnsureJoined(chatId, ownerId, cancellationToken)
-                    .ConfigureAwait(false);
+                // If chat is created with possibility to join anonymous authors, then join owner is anonymous author.
```
Comment (alexis-kochetov) [inline src/dotnet/Chat.Service/ChatsBackend.cs]: as

**Review by alexis-kochetov — APPROVED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/ContactSelect/ContactSelect.razor [hunk 2]
```diff
@@ -20,13 +21,31 @@
...
+    [Parameter] public ChatListOrder ChatListOrder { get; set; } = ChatListOrder.ByOwnUpdateTime;
     [Parameter, EditorRequired] public SearchPhrase SearchPhrase { get; set; } = null!;
     [Parameter] public IReadOnlyCollection<ChatId> ContactsToExclude { get; set; } = Array.Empty<ChatId>();
     [Parameter] public EventCallback<IReadOnlyCollection<ChatId>> SelectionChanged { get; set; }
+    [Parameter] public EventCallback<IReadOnlyCollection<ChatInfo>> ChatSelectionChanged { get; set; }
+
+    protected override void OnInitialized() {
+        base.OnInitialized();
+        _model = ContextModel ?? new ContactSelectContext.Model();
+        _model.Changed += async (_, _) => {
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ContactSelect/ContactSelect.razor:41]: can event handler's code throw?

File: src/dotnet/Chat.UI.Blazor/Components/NewChat/NewChatModalMembers.razor [hunk 3]
```diff
@@ -0,0 +1,69 @@
+@using ActualChat.Search
+@inherits FusionComponentBase
+@namespace ActualChat.Chat.UI.Blazor.Components
+
+<FormBlock>
+    <ContactSelectContext ChatSelectionChanged="OnSelectionChanged">
+        <SearchBox
+            Placeholder="Whom would you like to Add"
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/NewChat/NewChatModalMembers.razor:8]: was capital A intentional?

**Review by frolyo — APPROVED** [review]

File: src/dotnet/Chat.Service/ChatsBackend.cs (hunk 1 above)
Comment (crui3er) [reply src/dotnet/Chat.Service/ChatsBackend.cs]: fixed

File: src/dotnet/Chat.UI.Blazor/Components/NewChat/NewChatModalMembers.razor (hunk 3 above)
Comment (crui3er) [reply src/dotnet/Chat.UI.Blazor/Components/NewChat/NewChatModalMembers.razor:8]: Yes, i made it like on sketches.
https://www.figma.com/file/EJxXOzem02zhRvuqsJIool/Actual-chat-Desktop?type=design&node-id=11419-299386&mode=design&t=9zNJgdB9Z4G7ZfWI-0

File: src/dotnet/Chat.UI.Blazor/Components/ContactSelect/ContactSelect.razor (hunk 2 above)
Comment (crui3er) [reply src/dotnet/Chat.UI.Blazor/Components/ContactSelect/ContactSelect.razor:41]: I think it's responsibility of event handler to handle exception. We never catch exception of event handlers inside components.

**Pull request merged**